js/ui/file_manager/ui.file_manager.item_list.js

```diff
@@ -0,0 +1,57 @@
+import { extend } from "../../core/utils/extend";
+
+import Widget from "../widget/ui.widget";
+
+const FILE_MANAGER_FILES_VIEW_CLASS = "dx-filemanager-files-view";
+
+class FileManagerItemListBase extends Widget {
+
+    _initMarkup() {
+        this.$element().addClass(FILE_MANAGER_FILES_VIEW_CLASS);
+    }
+
+    _getDefaultOptions() {
+        return extend(super._getDefaultOptions(), {
+            selectionMode: "single",
+            onGetItems: null,
+            onError: null,
+            onSelectedItemOpened: null,
+            getItemThumbnail: null
+        });
+    }
+
+    _getItems() {
+        const getItemsMethod = this.option("onGetItems");
+        return getItemsMethod();
+    }
+
+    _raiseOnError(error) {
+        this._raiseEvent("Error", error);
+    }
+
+    _raiseSelectedItemOpened(item) {
+        this._raiseEvent("SelectedItemOpened", item);
+    }
+
+    _raiseEvent(eventName, arg) {
+        const fullEventName = "on" + eventName;
+        const handler = this.option(fullEventName);
```

Comment:
Use the _createActionByOption instead of handler

Reply:
Is it necessary for the internal widget?

Reply:
Even an internal widget should have consistent API and use common parts.